internal/config/config.go

```diff
@@ -2,152 +2,223 @@ package config
+	// an existing config. But we will leave it around for
+	// gopass fsck to (optionaly) clean it up.
+	if nm := os.Getenv("GOPASS_CONFIG_NO_MIGRATE"); !gitconfig.HasGlobalConfig() && nm == "" {
+		if err := migrateConfigs(); err != nil {
+			debug.Log("failed to migrate: %s", err)
+		}
+	}
+
+	// load the global config to get the root path
+	c.root = newGitconfig().LoadAll("")
+	c.root.NoWrites = noWrites
+
+	rootPath := c.root.Get("mounts.path")
+	if rootPath == "" {
+		if err := c.SetPath(PwStoreDir("")); err != nil {
+			debug.Log("failed to set path: %s", err)
+		}
+	}
+	// load again, this might add a per-store config from the root store
+	c.root.LoadAll(rootPath)
+	c.root.NoWrites = noWrites
+
+	if rootPath := c.root.Get("mounts.path"); rootPath == "" {
+		if err := c.SetPath(PwStoreDir("")); err != nil {
+			debug.Log("failed to set path: %s", err)
+		}
+	}
+
+	// set global defaults
+	c.root.Preset = gitconfig.NewFromMap(defaults)
+
+	for _, m := range c.Mounts() {
+		c.cfgs[m] = newGitconfig().LoadAll(c.MountPath(m))
+		c.cfgs[m].NoWrites = noWrites
+	}
+
 	return c
 }
 
-// SetConfigValue will try to set the given key to the value in the config struct.
-func (c *Config) SetConfigValue(key, value string) error {
-	if err := c.setConfigValue(key, value); err != nil {
-		return err
+// Get returns the given key from the root config.
+func (c *Config) Get(key string) string {
+	return c.root.Get(key)
+}
+
+// GetM returns the given key from the mount or the root config if mount is empty.
+func (c *Config) GetM(mount, key string) string {
+	if mount == "" {
+		return c.root.Get(key)
+	}
+
+	if cfg := c.cfgs[mount]; cfg != nil {
+		return cfg.Get(key)
 	}
 
-	return c.Save()
+	return ""
 }
 
-// setConfigValue will try to set the given key to the value in the config struct.
-func (c *Config) setConfigValue(key, value string) error {
-	value = strings.ToLower(value)
-	o := reflect.ValueOf(c).Elem()
-	for i := 0; i < o.NumField(); i++ {
-		jsonArg := o.Type().Field(i).Tag.Get("yaml")
-		if jsonArg == "" || jsonArg == "-" {
-			continue
-		}
-		if jsonArg != key {
-			continue
-		}
-		f := o.Field(i)
-		switch f.Kind() { //nolint:exhaustive
-		case reflect.String:
-			f.SetString(value)
-
-			return nil
-		case reflect.Bool:
-			switch {
-			case value == "true" || value == "on":
-				f.SetBool(true)
-
-				return nil
-			case value == "false" || value == "off":
-				f.SetBool(false)
-
-				return nil
-			default:
-				return fmt.Errorf("not a bool: %s", value)
-			}
-		case reflect.Int:
-			iv, err := strconv.Atoi(value)
-			if err != nil {
-				return fmt.Errorf("failed to convert %q to integer: %w", value, err)
-			}
-			f.SetInt(int64(iv))
-
-			return nil
-		default:
-			continue
-		}
+// GetBool returns true if the value of the key evaluates to "true".
+// Otherwise it returns false.
+func (c *Config) GetBool(key string) bool {
+	if strings.ToLower(strings.TrimSpace(c.Get(key))) == "true" {
```

Comment:
Do we really need the `strings.ToLower(strings.TrimSpace` here? Why not on the `Get` side?

Reply:
It's being applied to the value. We only want this here.